app/models/restart_signal_handler.rb

```diff
@@ -1,11 +1,18 @@
 # JobQueue locks a mutex, hence the need for a separate SignalHandler thread
 # Self-pipe is also best practice, since signal handlers can themselves be interrupted
-class SignalHandler
+class RestartSignalHandler
+  class << self
+    alias listen new
```

Comment:
You're pretty much arguing semantics, it was still very clear what was happening before. The signal handler knows how to read from its own pipe and handle that input. The actual invocation is "customizable" and hence outside the signal handler.